tests/integration/client/sdk/v1/test_metadata_properties.py

```diff
@@ -53,3 +53,30 @@ async def test_delete_metadata_property(self, role: UserRole, db: "AsyncSession"
 
         await db.refresh(dataset, attribute_names=["metadata_properties"])
         assert len(dataset.metadata_properties) == 0
+
+    @pytest.mark.parametrize("role", [UserRole.owner, UserRole.admin])
+    async def test_update_metadata_property(self, role: UserRole, db: "AsyncSession") -> None:
+        dataset = await DatasetFactory.create()
+        await TermsMetadataPropertyFactory.create(dataset=dataset)
+        await FloatMetadataPropertyFactory.create(dataset=dataset)
+        await IntegerMetadataPropertyFactory.create(dataset=dataset)
+        user = await UserFactory.create(role=role, workspaces=[dataset.workspace])
+
+        api = Argilla(api_key=user.api_key, workspace=dataset.workspace.name)
+
+        await db.refresh(dataset, attribute_names=["metadata_properties"])
+        assert len(dataset.metadata_properties) == 3
+
+        for metadata_property in dataset.metadata_properties:
+            response = update_metadata_property(
+                client=api.client.httpx, id=metadata_property.id, title="new-title", visible_for_annotators=False
+            )
+            assert response.status_code == 200
+            assert response.parsed.title == "new-title"
+            assert response.parsed.visible_for_annotators is False
+
+        await db.refresh(dataset, attribute_names=["metadata_properties"])
+        assert len(dataset.metadata_properties) == 3
+        for metadata_property in dataset.metadata_properties:
+            assert metadata_property.title == "new-title"
+            assert metadata_property.visible_for_annotators is False
```

Comment:
Avoid using test factories for integration tests. Also, I think this test is mostly covered in the `test_medatata.py` module

If you want to check API call details, you can create a unit test with a mocked HTTP client to check the request send

Reply:
But note that's a Python SDK test, not a Python client test, shouldn't we keep the factories then?